pos_customer_wallet/models/res_partner.py

```diff
@@ -34,6 +34,26 @@ def get_wallet_balance_bank_statement_line(self, all_partner_ids):
         self.env.cr.execute(query, where_clause_params)
         return self.env.cr.dictfetchall()
 
+    @api.model
+    def get_wallet_balance_pos_order_line(self, all_partner_ids):
+        # Note the current algorithm could not work in a multi company
+        # context, for partner shared in many companies.
+        # TODO : ask to carmen how `_where_calc` works, and how to make
+        # join.
```

Comment:
The algorithm for computing the wallet balance is borrowed from `ResPartner._invoice_total` in the `account`. This should probably be added as a comment both here in `pos_customer_wallet` and in `account_customer_wallet`.

I do not know more than you do on how `_where_calc` works. I just know that it does what needs doing, and is more performant (I think?) than the previous implementation. The change to the `_invoice_total` implementation was done in 851a2b8e12bc6f30eae4c9d75ab667e832adc291.

If `_where_calc` is black magic. We can adjust the algorithm again to something more legible.

Reply:
I did magic in [33f1a9f](https://github.com/coopiteasy/addons/pull/257/commits/33f1a9f46e843489119f33879fc5b34db75aed00). It works!